elephant/joint_isi_dithering_class.py

```diff
@@ -0,0 +1,462 @@
+'''
+Module to generate surrogates of spike trains by using the joint-ISI dithering.
```

Comment:
Few comments:
- I think that the joint-isi dithering class should be included in the spike train surrogates module
- I also think that the structure should be similar to the spiketrain surrogate module
I don't know if this is a massive job though

Reply:
1. I also added it to the surrogate method, as you may have seen. And there you can access it as the other surrogate methods (which for the user means less freedom). 
But I liked to give this space to the JOint-ISI. first that it is possible to separate preprocessing and processing as we will use it in SPADE. Also because it makes it much easier to investigate the properties changing the parameters on the fly.

Reply:
I am not really sure I understand the argument to (1) as such. Although I didn't go through the code thoroughly, I understand the reason for the class implementation, and the solution to have both a class and a normal function wrapper is good. However, having a separate module for this one class seems to me massive overkill, the class should be part of the surrogates module.

Reply:
I understand your arguments! Then I would add it as well to the surrogate modules, keeping the class structure, as @mdenker says

Reply:
@stellalessandra @mdenker No I placed the class inside the module of the surrogates.